collections/_posts/2022-11-28-CurriculumLearningByDIH.md

```diff
@@ -0,0 +1,117 @@
+    venue:   "NeurIPS 2020"
+pdf: "https://dl.acm.org/doi/pdf/10.5555/3495724.3496445"
+---
+
+# Notes
+
+* Code is available [here](https://github.com/tianyizhou/DIHCL)
+
+# Highlights
+
+* This paper proposes a new difficulty metric for curriculum learning called Dynamic Instantaneous Hardness (DIH).
+* Its purpose is to speed-up learning and to improve the results.
+* It is only tested on classification but might be extended to other applications.
+&nbsp;
+
+# Introduction
+
+Curriculum learning[^1] consists in presenting the right examples in the right order at the right time during training to enhance the learning process, just as teachers would do with their students. To that end, it is necessary to define a concept of _hardness_. It can be related to the different classes -in the case of classification- or inherent to the examples, such as specific shapes. One approach, called self-paced learning[^2], selects some examples at each epoch based on instantaneous feedback from the model. However, it does not take into account the training history of each sample, motivating the introduction of DIH.
+
+# Methods
+
+## Dynamic Instantaneous Hardness
+
+The goal of the measure is to get the usefulness of each sample in the future in order to select them wisely. The Dynamic Instantaneous Hardness (DIH) is defined as the exponential moving average of instantaneous hardness measures of a sample over time, the instantaneous hardness $$a_t(i)$$ being a measure retrieved from the epoch that has just been computed.
+
+$$
+r_{t+1}(i) = \left\{
+  \begin{array}
+    1 \gamma*a_t(i) + (1-\gamma)*r_{t}(i) \quad if \quad i \in S_t \\
+    r_t(i) \quad else
+  \end{array}
+  \right.
+$$
+
+$$ \gamma \in [0,1] $$ is a discount factor and $$S_t$$ is the set of samples used for training at time $$t$$.
+
+The three instantaneous hardness used to calculate DIH and compared are:
+* the loss : $$l(y_i,F(x_i; w_t))$$, where $$l(·, ·)$$ is a standard loss function and $$F(·; w)$$ is the model where $$w$$ are the model parameters
+* the loss change between two consecutive time steps : $$ \|l(y_i,F(x_i; w_t))-l(y_i,F(x_i; w_{t-1}))\| $$
+* the prediction flip (the 0-1 indicator of whether the prediction correctness changes) between two consecutive time steps : $$ \| 𝟙[\hat{y}^t_i = y_i]- 𝟙[\hat{y}^{t-1}_i = y_i]\| $$, where $$\hat{y}^t_i$$ is the prediction of sample $$i$$ in step $$t$$, e.g., $$ \arg\max_j F(x_i; w_t)[j]$$ for classification.
+
+![](/collections/images/CL_DIH/DIH_vs_SPL.jpg)
+
+Compared to DIH, instantaneous hardness is i) much less stable during training and ii) requires extra training and inference steps by a model over all the samples.
+
+
+## Properties
+
+DIH can vary a lot between different samples facilitating their discrimination, and so helping to select the most informative samples. When it is smaller, the samples are more memorable, i.e. easy to learn, while a larger DIH means that the samples are harder to retain.
```

Comment:
Just a slight suggestion on how to restructure the sentence so that it flows better :wink: 
```suggestion
DIH can vary a lot between different samples, helping to discriminate between them to select the most informative samples. When it is smaller, the samples are more memorable, i.e. easy to learn, while a larger DIH means that the samples are harder to retain.
```